### PR TITLE
make es6ify idempotent

### PR DESCRIPTION
**Pull request by hkollmann**
The problem is the call of babelCore.transform. This will change output during repeated calls.
Implement an modfied idea from @derrel. 

while (true) {
        result = babelCore.transform(src, config);
        if (result.code === src) {
          break;
        }
        src = result.code;
      }

### CODE REVIEW

**Review by johnspackman — APPROVED** [review]
Brilliant :)

Approving, just wondered if it works on the pre-es6ified source?

**Review by cboulanger — APPROVED** [review]

File: source/class/qx/tool/compiler/Es6ify.js
Comment (derrell) [inline source/class/qx/tool/compiler/Es6ify.js:178]: This looks good... but is there any chance that it could get into a state where it toggles back and forth between two results, or otherwise doesn't stabilize? Maybe add a maximum number of iterations (10? 50?) as a "just in case"? I'll approve this, but I do have a bit of concern.

**Review by derrell — APPROVED** [review]

File: source/class/qx/tool/compiler/Es6ify.js
Comment (derrell) [reply source/class/qx/tool/compiler/Es6ify.js:178]: Maybe even add a printed warning if it ever does hit the maximum count, so that we can look into what's causing it?

Comment (oetiker) [reply source/class/qx/tool/compiler/Es6ify.js:178]: yep, having an escape hatch sounds like a good thing ...

File: source/class/qx/tool/compiler/Es6ify.js
Comment (oetiker) [inline source/class/qx/tool/compiler/Es6ify.js:172]:
```suggestion
      let cycleCount = 0;
      while (true) {
```

Comment (derrell) [reply source/class/qx/tool/compiler/Es6ify.js:172]: lol. I'll assume you pressed the green button before you were finished typing in your suggestion...

**Comment by hkollmann** [comment]
@johnspackman : Yes it did. Tested with framework code.

**Comment by hkollmann** [comment]
Thought about an exit strategy after launching the PR...

**Review by oetiker — APPROVED** [review]

**Comment by hkollmann** [comment]
After merging this we have rerun the es6ify process...

**Review by derrell — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Pull request merged**
